lib/util.js

```diff
@@ -3,26 +3,24 @@
  */
 
 /**
- * A utility function to create a verification suite.
+ * A utility function to create a key type handler.
  *
  * @param {object} options - Options hashmap.
- * @param {Function} options.generate - A method that generates a KeyPair with
- *   an optional deterministic seed.
- * @param {Function} options.from - A method that creates a verification KeyPair
+ * @param {Function} options.from - A method that creates a KeyPair
  *   from an existing serialized key.
+ * @param {Function} options.fromFingerprint - A method that creates a KeyPair
+ *   from a key fingerprint.
  *
- * @returns {object} - Verification suite.
+ * @returns {object} - Keytype handler.
  */
-export function createVerificationSuite({generate, from}) {
+export function createKeyTypeHandler({from, fromFingerprint}) {
   return {
-    async from({...keyPair}) {
-      return from({...keyPair});
-    },
-    async fromFingerprint({fingerprint}) {
-      return from({publicKeyMultibase: fingerprint});
-    },
-    async generate({seed, ...keyPairOptions}) {
-      return generate({seed, ...keyPairOptions});
+    async from({publicKeyMultibase, ...keyPair}) {
+      if(publicKeyMultibase) {
+        return fromFingerprint({fingerprint: publicKeyMultibase});
+      } else {
+        return from({...keyPair});
+      }
```

Comment:
Ok, now that we've removed some of the things that didn't quite abstract right (like `generate`) ... it's really made it more clear how much complexity we could cut away. For instance, with all the private key stuff gone (which shouldn't have been part of a DID document anyway) ... it's become much more clear that all we need to register a `fromPublicKeyMultibase` function. So our `keyTypeHandler` isn't this complex thing, it's actually just that one function.

So we should rework things a little bit to just require that function. It's unfortunate that it's such a long name, so perhaps we can contextualize this and just use: `fromMultibase`. After all, DID keys all use a multibase representation of the key. I'll make this suggestion on `use`.